### PR TITLE
Added a precommit hook for linting python code

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Add a precommit hook for linting python code
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/442
 
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -140,9 +148,5 @@ Run from `bundle-workflow` before making pull requests.
 ```
 cd bundle-workflow
 
-git status -s | grep -e "[MA?]\s.*.py" | cut -c4- | xargs pipenv run black
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:143]: The pre-commit hook will run the checks, but doesn't auto-fix anything (and shouldn't). Let's put back the cheatsheet to fix things: maybe say "The pre-commit hook checks for ..." and "Auto-fix format and sort imports by running  `... run black`, and `isort .`"?

File: bundle-workflow/.pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+default_stages: [commit]
+default_language_version:
+  python: python3.7
+repos:
+- repo: local
+  hooks:
```
Comment (dblock) [inline bundle-workflow/.pre-commit-config.yaml:6]: Should we also specify `files: ^bundle-workflow/` to avoid running the hook on everything else?

**Review by dblock — CHANGES_REQUESTED** [review]

File: bundle-workflow/.pre-commit-config.yaml (hunk 2 above)
Comment (owaiskazi19) [reply bundle-workflow/.pre-commit-config.yaml:6]: Although I have made the entry to `bundle-workflow` but yeah adding `files` will be a better idea on top of that

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -140,9 +148,14 @@ Run from `bundle-workflow` before making pull requests.
 ```
 cd bundle-workflow
 
+The pre-commit hook checks for imports, type, style and test
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:151]: Don't wrap the line.
Add a period at the end of the sentence.

File: DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -140,9 +148,14 @@ Run from `bundle-workflow` before making pull requests.
...
+pre-commit run --all-files
+
+Auto-fix format and sort imports by running
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:154]: flake8, mypy and pytest don't autofix anything and should be removed now that the precommit hook does the checks

**Review by dblock — CHANGES_REQUESTED** [review]
You'll need to rebase and some minor stuff.

**Review by dblock — APPROVED** [review]
Thanks for taking care of this!

**Review by saratvemulapalli — APPROVED** [review]
This is nice!

**Pull request merged**